HeterogeneousCore/SonicTriton/src/TritonClient.cc

```diff
@@ -79,17 +79,29 @@ TritonClient::TritonClient(const edm::ParameterSet& params)
   if (!msg_str.empty())
     throw cms::Exception("ModelErrors") << msg_str;
 
+  const std::vector<edm::ParameterSet>& inputConverterDefs = params.getParameterSetVector("inputConverters");
+  std::unordered_map<std::string,std::string> inConvMap;
+  for (const auto& converterDef : inputConverterDefs) {
+    inConvMap[converterDef.getParameter<std::string>("inputName")] = converterDef.getParameter<std::string>("converterName");
+  }
+
   //setup input map
   std::stringstream io_msg;
   if (verbose_)
     io_msg << "Model inputs: "
            << "\n";
   inputsTriton_.reserve(nicInputs.size());
   for (const auto& nicInput : nicInputs) {
-    const auto& iname = nicInput.name();
+    const std::string iname_full = nicInput.name();
+    const auto& iname = iname_full.find("_DataConverter:") != std::string::npos ? iname_full.substr(0,iname_full.find("_DataConverter:")) : iname_full;

```

Comment:
here's how I would reduce duplication:

1. define a static function of TritonClient:
    ```cpp
    static std::pair<std::string, std::string> splitNameConverter(const std::string& fullname) {
      static const std::string indicator("_DataConverter:");
      size_t dcpos = fullname.find(indicator);
      if (dcpos != std::string::npos)
        return {fullname.substr(0,dcpos), fullname.substr(dcpos+indicator.size())};
      else
        return {fullname, ""};
    }
    ```
2. use the static function here:
    ```cpp
    const auto& [iname, iconverter] = TritonClient::splitNameConverter(iname_full);
    ```
3. pass `iconverter` to `curr_input.defaultConverter()` below (& do the same for the outputs further down)
4. `defaultConverter()` just has to check if the input parameter is non-empty, otherwise it tries to get a standard converter (as currently implemented)
5. `inConvMap` can be removed entirely (maybe not right away if you still want to test things)

Reply:
Ok, yes, this is cleaner, Ill make this change now, thanks. Regarding 5), I think we still want to let the user define their own converter. For example, in the case of FACILE the server would be configured to request FloatToApFixed16F6 as the converter, but if a dedicated FACILE for FPGA producer is written, it can use the same server and would need to use a different converter.

Reply:
I can imagine there might be cases where the user wants to provide an alternate converter, but I'm confused about your example. Why would a dedicated producer need to use a different converter for the same server?

Reply:
Im envisioning a dedicated FPGA producer that is aware of the `ap_fixed<16,6>` type that is needed by the server, so it places each input in the data vector already converted, so the converter is basically just a default `UINT16` converter.

Reply:
Ah okay, I understand now.